Sources/StreamCore/Query/Filter+Local.swift

```diff
@@ -0,0 +1,237 @@
+        switch filterRawJSONValue {
+        case .bool(let exists): exists ? !localRawJSONValue.isNil : localRawJSONValue.isNil
+        default: false
+        }
+    }
+
+    static func isEqual(_ localRawJSONValue: RawJSON, _ filterRawJSONValue: RawJSON) -> Bool {
+        switch (localRawJSONValue, filterRawJSONValue) {
+        case (.string(let lhsValue), .string(let rhsValue)): lhsValue.localizedCaseInsensitiveCompare(rhsValue) == .orderedSame
+        default: localRawJSONValue == filterRawJSONValue
+        }
+    }
+    
+    static func isGreater(_ localRawJSONValue: RawJSON, _ filterRawJSONValue: RawJSON) -> Bool {
+        switch (localRawJSONValue, filterRawJSONValue) {
+        case (.number(let lhsValue), .number(let rhsValue)): lhsValue > rhsValue
+        case (.string(let lhsValue), .string(let rhsValue)): lhsValue.localizedCaseInsensitiveCompare(rhsValue) == .orderedDescending
+        default: false
+        }
+    }
+    
+    static func isGreaterOrEqual(_ localRawJSONValue: RawJSON, _ filterRawJSONValue: RawJSON) -> Bool {
+        switch (localRawJSONValue, filterRawJSONValue) {
+        case (.number(let lhsValue), .number(let rhsValue)): lhsValue >= rhsValue
+        case (.string(let lhsValue), .string(let rhsValue)): lhsValue.localizedCaseInsensitiveCompare(rhsValue) != .orderedAscending
+        default: false
+        }
+    }
+    
+    static func isLess(_ localRawJSONValue: RawJSON, _ filterRawJSONValue: RawJSON) -> Bool {
+        switch (localRawJSONValue, filterRawJSONValue) {
+        case (.number(let lhsValue), .number(let rhsValue)): lhsValue < rhsValue
+        case (.string(let lhsValue), .string(let rhsValue)): lhsValue.localizedCaseInsensitiveCompare(rhsValue) == .orderedAscending
+        default: false
+        }
+    }
+    
+    static func isLessOrEqual(_ localRawJSONValue: RawJSON, _ filterRawJSONValue: RawJSON) -> Bool {
+        switch (localRawJSONValue, filterRawJSONValue) {
+        case (.number(let lhsValue), .number(let rhsValue)): lhsValue <= rhsValue
+        case (.string(let lhsValue), .string(let rhsValue)): lhsValue.localizedCaseInsensitiveCompare(rhsValue) != .orderedDescending
+        default: false
+        }
+    }
+    
+    static func autocomplete(_ localRawJSONValue: RawJSON, _ filterRawJSONValue: RawJSON) -> Bool {
+        switch (localRawJSONValue, filterRawJSONValue) {
+        case (.string(let localStringValue), .string(let filterStringValue)):
+            return localStringValue.range(of: filterStringValue, options: [.anchored, .caseInsensitive], locale: .autoupdatingCurrent) != nil
+        default:
+            return false
+        }
+    }
+    
+    static func query(_ localRawJSONValue: RawJSON, _ filterRawJSONValue: RawJSON) -> Bool {
+        switch (localRawJSONValue, filterRawJSONValue) {
+        case (.string(let localStringValue), .string(let filterStringValue)):
+            return localStringValue.localizedCaseInsensitiveContains(filterStringValue)
```

Comment:
Doubt on my side, since this doesn't 100% match the backend implementation.

As far as I understand, the backend is doing Postgres full text search, which has its own behavior. For instance, given an activity with text "Example", I'm getting a match for the plural "examples", but not for "exa".

Buuut, we don't have a way to filter like the backend. Was it discussed to fallback to contains because of this?